File: decorator/java/IDataSource.java
package Decorator;

// Üst sınıfların uygulayacağı arayüzdür.
// Yapılacak temel işlemler tanımlanmıştır.
// UML diyagramındaki Component arayüzüne denk gelmektedir.
interface IDataSource
{
    String getFileName();
    void writeData(Object data);
    void readData();
}

File: decorator/kotlin/IDataSource.kt
package Decorator

// Üst sınıfların uygulayacağı arayüzdür.
// Yapılacak temel işlemler tanımlanmıştır.
// UML diyagramındaki Component arayüzüne denk gelmektedir.
internal interface IDataSource {
    val fileName: String
    fun writeData(data: Any)
    fun readData()
}
